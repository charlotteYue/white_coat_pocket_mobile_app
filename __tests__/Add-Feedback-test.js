import 'react-native';
import React from 'react';
import Feedback from '../pages/feedback';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

global.fetch = jest.fn(() => new Promise(resolve => resolve()));
jest.mock('react-native-gesture-handler', () => {});

it('renders add feedback component correctly', async () => {
  const tree = renderer.create(<Feedback />).toJSON();
  expect(tree).toMatchSnapshot();
});