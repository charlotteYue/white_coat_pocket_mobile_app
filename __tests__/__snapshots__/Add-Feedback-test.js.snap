// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders add feedback component correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#1565C0",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1565C0",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "left": 0,
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        forwardedRef={[Function]}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flex": 1,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "left": 0,
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
      >
        <Text
          allowFontScaling={false}
          onPress={[Function]}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "color": "#007aff",
                  "fontSize": 24,
                  "marginLeft": 16,
                  "marginRight": 16,
                  "paddingTop": 2,
                },
                Object {
                  "color": "white",
                },
              ],
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 3,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "alignItems": "center",
            "color": "#ffffff",
            "fontSize": 20,
            "fontWeight": "bold",
            "justifyContent": "center",
          }
        }
      >
        White Coat Pocket Guide
      </Text>
    </View>
    <View>
      <View
        accessible={true}
        focusable={false}
        forwardedRef={[Function]}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
      >
        <Text
          allowFontScaling={false}
          onPress={[Function]}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "color": "#007aff",
                  "fontSize": 24,
                  "marginLeft": 16,
                  "marginRight": 16,
                  "paddingTop": 2,
                },
                Object {
                  "color": "white",
                },
              ],
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    style={
      Object {
        "marginBottom": 60,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#1565C0",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 20,
              "fontWeight": "bold",
              "padding": 20,
            }
          }
        >
           Submit a Feedback 
        </Text>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#E3F2FD",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginTop": 10,
              }
            }
          >
             Your Email Address
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            onChangeText={[Function]}
            placeholder="Email Address"
            rejectResponderTermination={true}
            style={
              Object {
                "backgroundColor": "#E3F2FD",
                "borderRadius": 10,
                "height": 40,
                "marginTop": 10,
                "paddingHorizontal": 10,
                "width": 250,
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <Text
            style={
              Object {
                "color": "#E3F2FD",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginTop": 10,
              }
            }
          >
            Your Feedback
          </Text>
          <TextInput
            allowFontScaling={true}
            minHeight={120}
            multiline={true}
            numberOfLines={null}
            onChangeText={[Function]}
            placeholder="Feedback"
            rejectResponderTermination={true}
            style={
              Object {
                "backgroundColor": "#E3F2FD",
                "borderRadius": 10,
                "height": 40,
                "marginTop": 10,
                "paddingHorizontal": 10,
                "width": 250,
              }
            }
            textAlignVertical="top"
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "marginTop": 50,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#1E88E5",
                  "borderColor": "#1976D2",
                  "borderRadius": 20,
                  "borderWidth": 5,
                  "height": 30,
                  "justifyContent": "center",
                  "width": 100,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#E3F2FD",
                      "fontSize": 15,
                      "fontWeight": "bold",
                      "maxWidth": 100,
                      "textAlign": "center",
                    }
                  }
                >
                  Submit
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
